### PR TITLE
K8s repair task

### PR DESCRIPTION
**Pull request by dkropachev**
https://trello.com/c/M1sCKVWL/2590-k8s-mgmtrepairkubernetes

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

**Comment by dkropachev** [comment]
Tested - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dmitry/job/dmitry-operator-byo-2/69

**Review by bentsi — COMMENTED** [review]
please run one of them manager master jobs on thir PR

**Comment by dkropachev** [comment]
Mgmt job tested here - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dmitry/job/centos-sanity-test/2
It has failed, but with the same reason as it is failingn on the master

**Comment by bentsi** [comment]
pre-commit failed
image

**Review by bentsi — APPROVED** [review]

File: sdcm/cluster.py [hunk 1]
```diff
@@ -2881,8 +2884,12 @@ def __init__(self, cluster_uuid=None, cluster_prefix='cluster', node_prefix='nod
         self.log = SDCMAdapter(LOGGER, extra={'prefix': str(self)})
         self.log.info('Init nodes')
         self.nodes = []
-        self.instance_provision = params.get('instance_provision')
-        self.params = params
+        if params:
+            self.instance_provision = params.get('instance_provision')
+            self.params = params
+        else:
+            self.instance_provision = None
```
Comment (fruch) [inline sdcm/cluster.py:2891]: when base cluster is not getting called with params ?

File: sdcm/nemesis.py [hunk 2]
```diff
@@ -1701,6 +1699,25 @@ def disrupt_mgmt_repair_cli(self):
             mgr_task.id, str(mgr_task.status))
         self.log.info('Task: {} is done.'.format(mgr_task.id))
 
+    @latency_calculator_decorator
+    def disrupt_mgmt_repair_kubernetes(self):
```
Comment (fruch) [inline sdcm/nemesis.py:1703]: I'm not sure why we need duplicate of this nemesis ? how it's different from the regular one ?

Comment (fruch) [reply sdcm/nemesis.py:1703]: is this temporary ?

Comment (dkropachev) [reply sdcm/nemesis.py:1703]: It is totally different nemesis, it tests operator feature.
Eventhou this feature uses scylla-manager under the hood, the target here is scylla-operator, not manager.

File: sdcm/cluster.py (hunk 1 above)
Comment (dkropachev) [reply sdcm/cluster.py:2891]: In my local test i'am doing it all the times.
At the same time we define params=None, so we allow it to be None, but not covering this case in __init__

File: sdcm/mgmt/cli.py [hunk 3]
```diff
@@ -6,20 +6,19 @@
 import json
```
Comment (fruch) [inline sdcm/mgmt/cli.py:6]: this file movement is going to break `siren-tests`
https://github.com/scylladb/siren-tests/blob/master/sct_plugin/cluster_cloud.py#L20

Comment (fruch) [reply sdcm/mgmt/cli.py:6]: I take that back, since `sdcm/mgmt/__init__.py` is importing it

File: sdcm/nemesis.py (hunk 2 above)
Comment (fruch) [reply sdcm/nemesis.py:1703]: I understand it, but the flow of the nemesis should be identical, now if we have thing we need to add to it, or fix in this flow.
we have two places.

we are using this nemesis also on cloud, as you've seen, without adding a new nemesis specifically for it.

anyhow that's for @bentsi to decide. 
I find it weird we duplicate those things, if there a current bug we can put and if in the right place based on the `cluster_backend`.

File: sdcm/cluster.py (hunk 1 above)
Comment (fruch) [reply sdcm/cluster.py:2891]: then you have bug somewhere no passing down the params correctly

File: sdcm/mgmt/operator.py [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+    @property
+    def mgmt_task_id(self) -> str:
+        return f'repair/{self.id}'
+
+
+@dataclass
+class ScyllaOperatorBackupTask(ScyllaOperatorTaskBaseClass):
+    # DC list of datacenter glob patterns, e.g. 'dc1', '!otherdc*' used to specify the DCs
+    # to include or exclude from backup.
+    dc: List[str] = None
```
Comment (fruch) [inline sdcm/mgmt/operator.py:93]: why we need to repeat all those and their docs ? it's not exactly clear to me. this is from the operator docs ? or from `sctool` docs ?

i.e. now we have multiple places to update on each `scylla-manager` new release (and they move around and add new flags, almost at any release)

File: sdcm/cluster.py (hunk 1 above)
Comment (fruch) [reply sdcm/cluster.py:2891]: I rather fix that, and assert params shouldn't be empty, you might be overwriting the `self.params` here, and breaking other code paths.

**Comment by fruch** [comment]
@dkropachev in general it looks o.k., I'll let @bentsi decide on the two maintainability issues there.
also regarding 
https://github.com/scylladb/scylla-cluster-tests/pull/3012#discussion_r549664344, I'm not sure how we didn't encounter that until now...

File: sdcm/nemesis.py (hunk 2 above)
Comment (bentsi) [reply sdcm/nemesis.py:1699]: @fruch agree, we need to have a new flavor of manager cluster for Operator that will have the same API so we won't duplicate the nemesis code. Dmitry, please improve.

Comment (dkropachev) [reply sdcm/nemesis.py]: redone

**Comment by dkropachev** [comment]
Tested: 
https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dmitry/job/dmitry-operator-byo-5/100
https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dmitry/job/sct-feature-test-backup/2/

File: sdcm/cluster.py (hunk 1 above)
Comment (dkropachev) [reply sdcm/cluster.py]: It is needed for scylla_manager_cluster, which is PodCluster that have no params

**Review by bentsi — APPROVED** [review]

**Pull request merged**